# config/*.yaml files are used to configure the camera parameters
---
orbbec_ros:
    camera_parameters:
        general:
            # Camera model. upport product models by referencing config/*.yaml
            camera_model: "femto_bolt"
            # Path to the configuration file. If empty, default parameters will be used
            config_file_path: "femto_bolt.yaml"

        deivce:
            camera_name: "camera"
            serial_number: ""
            usb_port: ""
            device_num: 1
            uvc_backend: "libuvc"
            product_id: ""
            connection_delay: 100
            time_domain: "global"
            enable_heartbeat: false
            enable_ldp: true

        color:
            enable_color: true
            color_qos: "default"
            color_camera_info_qos: "default"
            color_width: 1280
            color_height: 720
            color_fps: 30
            color_format: "MJPG"
            enable_color_auto_white_balance: true
            color_white_balance: -1
            color_auto_ae:
                enable_color_auto_exposure: true
                color_ae_max_exposure: -1
            color_manual_ae:
                color_exposure: -1
                color_gain: -1
                color_brightness: -1

        depth:
            enable_depth: true
            depth_qos: "default"
            depth_camera_info_qos: "default"
            depth_width: 640
            depth_height: 576
            depth_fps: 30
            depth_format: "Y16"
            depth_registration: false
            align_mode: "SW"
        ir:
            enable_ir: true
            ir_qos: "default"
            ir_camera_info_qos: "default"
            ir_width: 640
            ir_height: 576
            ir_fps: 30
            ir_format: "Y16"

        ir_parameters:
            enable_ir_auto_exposure: true
            ir_ae_max_exposure: -1
            ir_exposure: -1
            ir_gain: -1
            ir_brightness: -1

        imu:
            enable_accel: false
            accel_rate: "200hz"
            accel_range: "4g"
            enable_gyro: false
            gyro_rate: "200hz"
            gyro_range: "1000dps"
            enable_sync_output_accel_gyro: true
            linear_accel_cov: 0.01
            angular_vel_cov: 0.01

        point_cloud:
            enable_point_cloud: false
            enable_colored_point_cloud: false

        advanced:
            enable_frame_sync: true
            enable_d2c_viewer: false

